src/include/rados/librados.hpp

```diff
@@ -1477,8 +1477,11 @@ inline namespace v14_2_0 {
     int get_pool_stats(std::list<std::string>& v,
                        std::string& category,
 		       std::map<std::string, stats_map>& stats);
-    /// check if pool has selfmanaged snaps
-    bool get_pool_is_selfmanaged_snaps_mode(const std::string& poolname);
+
+    /// check if pool has or had selfmanaged snaps
+    bool get_pool_is_selfmanaged_snaps_mode(const std::string& poolname)
+      __attribute__ ((deprecated));
+    int pool_is_in_selfmanaged_snaps_mode(const std::string& poolname);
```

Comment:
Unless I'm misunderstanding, this is removing an existing function without a deprecation period.

Do we version librados? (I think we do?) I know we're not good about revving it, but this should increment the version.

I realize the old version is unsafe, but we should at least make it a pass-thru to the new safe implementation, perhaps tweaking the return value on pool ENOENT cases.

Reply:
> we should at least make it a pass-thru to the new safe implementation, perhaps tweaking the return value on pool ENOENT cases.

This is exactly what is done here:
```
// deprecated, use pool_is_in_selfmanaged_snaps_mode() instead
bool librados::Rados::get_pool_is_selfmanaged_snaps_mode(const std::string& pool)
{
  return client->get_pool_is_selfmanaged_snaps_mode(pool);
  // errors are ignored, prone to false negative results
  return client->pool_is_in_selfmanaged_snaps_mode(pool) > 0;
}
```

Reply:
*sigh* sorry for the noise, I saw this on my phone and missed the librados_cxx change and got confused by the RadosClient one.